journal/db.py

```
@@ -0,0 +1,106 @@
+import json
+from collections import Counter
+from dataclasses import asdict
+from datetime import datetime
+from pathlib import Path
+from typing import Final, List
+from uuid import uuid4
+
+from models import JournalEntry
+
+DB_FILE = Path("journal.json")
+DASH_LENGTH: Final[int] = 80
+
+
+class JournalDatabase:
+    def __init__(self, db_file: Path):
+        self.db_file = db_file
+        if not self.db_file.exists():
+            self.db_file.touch(exist_ok=True)
+
+    def load_entries(self) -> List[JournalEntry]:
+        """Load journal entries from the JSON file."""
+        entries: List[JournalEntry] = []
+        with self.db_file.open("r", encoding="utf-8") as f:
+            try:
+                data = json.load(f)
+                return [JournalEntry(**entry) for entry in data]
+            except json.JSONDecodeError:
+                return entries
```

I wonder if we want to reraise an exception here?